app/Http/Controllers/CategoryController.php

```diff
@@ -4,73 +4,66 @@
 
 use App\Http\Requests\AddCategoryRequest;
 use App\Models\Category;
+use App\Repositories\Category\CategoryRepositoryInterface;
 use Illuminate\Http\Request;
 
 class CategoryController extends Controller
 {
+    protected $categoryRepo;
+
+    public function __construct(CategoryRepositoryInterface $categoryRepo)
```

Comment:
1. Tại sao bn sử dụng Repository? Ưu điểm của việc sử dụng Repository là gì so với cách dùng Model thông thường?
2. Tại sao bn sử dụng Interface ở đây thay vì CategoryRepository? Làm như ở đây có ưu điểm gì ko?

Reply:
1.Mình sử dụng repository để có thể dễ dàng viết unit test hơn.  Khi sử dụng repository nếu như có thay đổi ở câu truy vấn dữ liệu thì thay vì việc phải tìm tất cả những chỗ trong controller sử dụng câu truy vấn đó để sửa thì ta chỉ cần sửa ở trong repository là được
2.Sử dụng interface thay vì CategoryRepository để áp dụng tính đa hình. Giả sử sau này cũng repository này nhưng sử dụng 1 hệ csdl khác thì thay vì việc phải sửa toàn bộ những chỗ sử dụng repository thì ta chỉ cần sửa trong file provider để laravel hiểu interface ta sử dụng là của class nào.